src/backend/kvm/builder.rs

```diff
@@ -0,0 +1,111 @@
+// SPDX-License-Identifier: Apache-2.0
+
+use super::config::Config;
+
+use std::convert::TryFrom;
+use std::sync::{Arc, RwLock};
+
+use crate::backend::kvm::mem::Region;
+use anyhow::{Error, Result};
+use kvm_bindings::bindings::kvm_userspace_memory_region;
+use kvm_bindings::fam_wrappers::KVM_MAX_CPUID_ENTRIES;
+use kvm_ioctls::{Kvm, VmFd};
+use lset::Span;
+use mmarinus::{perms, Map};
+use sallyport::elf::pf::kvm::SALLYPORT;
+use sallyport::Block;
+use std::collections::VecDeque;
+use std::mem::size_of;
+use std::num::NonZeroUsize;
+use x86_64::VirtAddr;
+
+pub struct Builder {
+    _cnfg: Config,
+    kvm_fd: Kvm,
+    vm_fd: VmFd,
+    regions: Vec<Region>,
+    sallyport_blocks: Option<Span<VirtAddr, NonZeroUsize>>,
+}
+
+impl TryFrom<super::config::Config> for Builder {
+    type Error = Error;
+
+    fn try_from(config: super::config::Config) -> Result<Self> {
+        let kvm_fd = Kvm::new().unwrap();
+        let vm_fd = kvm_fd.create_vm().unwrap();
+        vm_fd.create_irq_chip().unwrap();
```

Comment:
`.unwrap()` is incorrect here. Let's return an error. I think you can just use `?` here...

Reply:
I'll remove the whole line... sorry, that was a left-over from debugging.